salt/metalk8s/addons/shell-ui/deployed/shell-ui.sls.in

```diff
@@ -0,0 +1,64 @@
+include:
+- .namespace
+
+# Todo see how to deal with workload plane config
+{%- from "metalk8s/addons/nginx-ingress-control-plane/control-plane-ip.sls"
+    import ingress_control_plane with context
+%}
+
+Create shell-ui deployment:
+  metalk8s_kubernetes.object_present:
+    - name: salt://{{ slspath }}/files/shell-ui-deployment.yaml
+    - template: jinja
+
+Create shell-ui service:
+  metalk8s_kubernetes.object_present:
+    - manifest:
+        apiVersion: v1
+        kind: Service
+        metadata:
+          name: shell-ui
+          namespace: metalk8s-ui
+          labels:
+            run: shell-ui
+        spec:
+          ports:
+          - port: 80
+            protocol: TCP
+            targetPort: 80
+          selector:
+            app: shell-ui
+          type: ClusterIP
+
+Create shell-ui ConfigMap:
+  metalk8s_kubernetes.object_present:
+    - manifest:
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: shell-ui
+          namespace: metalk8s-ui
+        data:
+          config.json: |
+            {
+              "docUrl": "/docs",
+              "oidc": {
+                "providerUrl": "/oidc",
+                "redirectUrl": "https://{{ ingress_control_plane }}/",
+                "clientId": "metalk8s-ui",
```

Comment:
So, the navbar will always use the same OIDC client, whichever UI includes it? Shouldn't we rename this client, then?

Reply:
As part of metalk8s UI I believe we should stick with this client. OIDC configuration are overridable through attributes on the navbar. This allow customisation for ui exposed on the workload plane for example.